### PR TITLE
test(evmstaking/keeper): add test cases for deposit

### PR DESCRIPTION
**Pull request by zsystm**
**coverage:**
- Increased test coverage from 0% to 94.4%
- **What's Covered**
  - Most application logics written on the Story side, excluding SDK.
- **What’s Not Covered (Edge Cases Related to the SDK)** 
  - Failed to set of map (= collections)
  - Staking keeper type assertion 

**changess**
- Added account keeper as a field of test suite (we need this for writing testing code)
- fix error message

issue: #64

### CODE REVIEW

File: client/x/evmstaking/keeper/deposit.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func (k Keeper) ProcessDeposit(ctx context.Context, ev *bindings.IPTokenStakingD
 	)
 
 	// Note that, after minting, we save the mapping between delegator bech32 address and evm address, which will be used in the withdrawal queue.
-	// The saving is done regardless of any error below, as the money is already minted and sent to the delegator, who can withdraw the minted amount.
+	// The saving is done regardless of any error below, as the money is already minted and sent to the delegator, who can deposit the minted amount.
```
[the remark this answers is not in the document]
Comment (zsystm) [reply client/x/evmstaking/keeper/deposit.go:70]: @limengformal 
oh, I don’t think it was intended to be changed. 
It seems like something got modified during the replacement, so I’ll check it.

[the remark this answers is not in the document]
Comment (zsystm) [reply client/x/evmstaking/keeper/deposit.go]: @limengformal 
I think `withdraw` is right description. I reverted the change. 
Thanks for mentioning it.

**Comment by zsystm** [comment]
@limengformal 
I rebased based on `upstream:main` and resolved conflicts.
While resolving conflicts, I moved the common testing helper function to `keeper_test.go` at https://github.com/piplabs/story/pull/71/commits/fc396ced669cea1edc4dc60166dcbcf8eda1a443 which is the 
 only change from the last version you reviewed.

**Review by limengformal — APPROVED** [review]
